src/main/scala/com/microsoft/hyperspace/index/rules/RuleUtils.scala

```diff
@@ -440,22 +440,6 @@ object RuleUtils {
     planForAppended
   }
 
-  private def extractBasePath(partitionSpec: PartitionSpec): Option[String] = {
-    if (partitionSpec == PartitionSpec.emptySpec) {
-      None
-    } else {
-      // For example, we could have the following in PartitionSpec:
-      //   - partition columns = "col1", "col2"
-      //   - partitions: "/path/col1=1/col2=1", "/path/col1=1/col2=2", etc.
-      // , and going up the same number of directory levels as the number of partition columns
-      // will compute the base path. Note that PartitionSpec.partitions will always contain
-      // all the partitions in the path, so "partitions.head" is taken as an initial value.
-      val basePath = partitionSpec.partitionColumns
-        .foldLeft(partitionSpec.partitions.head.path)((path, _) => path.getParent)
-      Some(basePath.toString)
-    }
-  }
-
```

Comment:
Note: moved to PathUtils

Reply:
@apoorvedave1 Could you rebase this change as #265 is merged?
We could use `Hyperspace.getContext(spark).sourceProviderManager.partitionBasePath(location)`.

Reply:
thanks @sezruby , fixed